### PR TITLE
:bug: controllerutil.RemoveFinalizer can occurs slice bounds out of range exception

### LINKED ISSUE #916: controllerutil.RemoveFinalizer could potentially occurs slice bounds out of range exception
func controllerutil.RemoveFinalizer(...) implementation could occurs slice bounds out of range exception even list contains more than one same entities

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #916

### CODE REVIEW

**Comment by dmvolod** [comment]
I signed it

**Comment by alexeldeib** [comment]
Can you please add a test case using the same finalizer twice to trigger this failure so we avoid regressions?

File: pkg/controller/controllerutil/controllerutil_test.go [hunk 1]
```diff
@@ -445,6 +445,12 @@ var _ = Describe("Controllerutil", func() {
 				controllerutil.RemoveFinalizer(deploy, testFinalizer)
 				Expect(deploy.ObjectMeta.GetFinalizers()).To(Equal([]string{}))
 			})
+
+			It("should remove all equal finalizers if present", func() {
+				deploy.SetFinalizers(append(deploy.Finalizers, testFinalizer, testFinalizer))
```
Comment (dmvolod) [inline pkg/controller/controllerutil/controllerutil_test.go:450]: @alexeldeib fixed there

**Comment by dmvolod** [comment]
@alexeldeib could you please to check CLA again. Thanks.

**Pull request merged**